frontend/tests/features/home/homePage.test.ts

```diff
@@ -11,62 +11,120 @@ describe("Homepage", () => {
     expect(await page.getByTestId("logo").count()).toBe(2);
     await expect(page.getByTestId(COLLECTION_LINK)).toBeVisible();
 
-    // check page scroll
-    expect(await scrollToPageBottom(page)).toBeTruthy;
+    await isPageScrollableToSeeSiteMap(page);
 
     // check collection link works
-    await Promise.all([
-      page.waitForNavigation(),
-      page.getByTestId(COLLECTION_LINK).click(),
-    ]);
-    expect(page.url()).toContain("collection");
-    // check page scroll
-    expect(await scrollToPageBottom(page)).toBeTruthy;
+    await page.getByTestId(COLLECTION_LINK).click();
+    await page.waitForURL("**" + ROUTES.COLLECTIONS);
+
+    await isGlobalLayoutWrapperScrollable(page);
   });
   test("Should render the ToS Page", async ({ page }) => {
     await goToPage(`${TEST_URL}${ROUTES.TOS}`, page);
 
-    await expect(page.locator("h1").getByText("Terms of Use")).toBeVisible();
+    await tryUntil(
```

Comment:
wrap assertion in `tryUntil` since sometimes the page is not loaded before the assertion is executed. Here and throughout

Reply:
OK